### ISSUE TITLE
useDebounceFn options add two key leading and trailing like lodash

### ISSUE THREAD

**Issue by MinMr** [vueuse/vueuse#3154]
### Clear and concise description of the problem

UseDebounceFn can only execute events after the delay ends. In fact, I would prefer to execute events before the delay starts

### Suggested solution

The third parameter options of useDebounceFn add two key leading and trailing to determine whether the event is executed before or after the delay starts

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by wo4188** [comment]
Yes, I agree. I found that useThrottleFn is like this, so why isn't useDebounceFn like this?

**Comment by Airkro** [comment]
Can anyone reopen this?
